File: trunk/java/JavaMercFramework/src/main/it/mintlab/desktopjava/mercframework/IFDispatcher.java
package it.mintlab.desktopjava.mercframework;

	/** 
	 This interface show Merc Node api for the Framework
	 
	*/
	public interface IFDispatcher
	{
		/** 
		 Give message to Merc Node
		 
		 <param id="message">message
		*/
		void letMessage(Message message);

		/** 
		 Kill the Merc
		 
		*/
		void killMerc();

		/**
		 *  Return Merc State
		 */
		boolean isMercAlive();

	}

File: trunk/net/NetMercFramework/IFDispatcher.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace it.mintlab.desktopnet.mercframework
{
    /// <summary>
    /// This Interface represent the dispatcher api for Framework
    /// </summary>
    internal interface IFDispatcher
    {
        /// <summary>
        /// Give message to Merc Node
        /// </summary>
        /// <param id="message">message</param>
        void letMessage(Message message);

        /// <summary>
        /// Kill the Merc
        /// </summary>
        void killMerc();

        /// <summary>
        /// Return Merc State
        /// </summary>
        /// <returns></returns>
        bool isMercAlive();

        
    }

}
